adapter/src/main/java/com/twilio/kudu/sql/rel/KuduToEnumerableRel.java

```diff
@@ -218,20 +218,22 @@ private Result executeQuery(EnumerableRelImplementor implementor, Prefer pref) {
     // If the output type has a Single column, inform the ScannerCallback.
     final Expression isSingleObject = Expressions.constant(physType.getRowType().getFieldCount() == 1);
 
+    List<Integer> sortedPkColumnIndexes = getPrimaryKeyColumnsInProjection(kuduImplementor.sortPkColumns,
+        kuduColumnIndices);
     final Expression sortedPrefixKeySelector;
-    if (!kuduImplementor.sortPkPrefixColumns.isEmpty()) {
-      sortedPrefixKeySelector = physType.generateCollationKey(kuduImplementor.sortPkPrefixColumns).left;
```

Comment:
`generateCollationKey` works only if you are sorting by 1 column, changed to use `generateAccessor` which can be used to compare keys when sorting by more than 1 column.